django/core/management/commands/makemessages.py

```diff
@@ -260,6 +260,15 @@ def add_arguments(self, parser):
             help="Don't write '#: filename:line' lines.",
         )
         parser.add_argument(
+            '--add-location', default=False, dest='add_location',
+            choices=('full', 'file', 'never'), const=True, nargs='?',
+            help="Control '#: filename:line' lines. The optional type can be either 'full', "
+                 "'file', or 'never'. If it is not given or 'full', it generates the lines "
+                 "with both file name and line number. If it is 'file', the line number "
+                 "part is omitted. If it is 'never', it completely suppresses the lines "
+                 "(same as --no-location).",
+        )
```

Comment:
I think you can simplify this expression removing the `default` and using `const='full'` instead of `const=True`:
```python
            '--add-location', dest='add_location',
            choices=('full', 'file', 'never'), const='full', nargs='?',
            help="Control '#: filename:line' lines. The optional type can be either 'full', "
                 "'file', or 'never'. If it is not given or 'full', it generates the lines "
                 "with both file name and line number. If it is 'file', the line number "
                 "part is omitted. If it is 'never', it completely suppresses the lines "
                 "(same as --no-location).",
 )
```

This would avoid having to treat the `options['add_location'] is True` below.